packages/nuxt/src/app/plugins/chunk-reload.client.ts

```diff
@@ -0,0 +1,23 @@
+import { defineNuxtPlugin } from '#app/nuxt'
+import { useRouter } from '#app/composables/router'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const router = useRouter()
+
+  let chunkError: TypeError | null = null
+  nuxtApp.hook('app:chunkError', ({ error }) => { chunkError = error })
```

Comment:
We could also catch this directly here:

```suggestion
  nuxtApp.hook('app:chunkError', ({ error }) => {
    if (nuxtApp._processingMiddleware) {
      // handle error here
    }
  })
```

I've not done that here in case the user has handled the error, but it's worth keeping in mind if someone has a better idea.